app/services/remote_identities/auto_create.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module RemoteIdentities
+  class AutoCreate
```

Comment:
❓ I have a wider question: What is the advantage here compared to emitting an event when the user logs in?

Reply:
I ask because the current implementation departs heavily from the use of events that the rest of the code base relies heavily (mildly?) on. It is not inherently wrong or bad, it is just different. 

Now that you mention the "login" events, maybe we should have a quick event storming session about the process of user <=> storage login. Why? We have this change, the ensure connection thingy, a couple of extra checks here and there etc. I'm getting the feeling that this entire process isn't very clear to us all.

Reply:
Sounds good to me.

Reply:
Oh, and that shouldn't block this PR at all. We can always go back to the drawing board with the knowledge we gain from the session.